packages/analytics-core/src/plugins/destination.ts

```diff
@@ -170,6 +176,11 @@ export class Destination implements DestinationPlugin {
       return;
     }
 
+    if (res.body.error.startsWith(INVALID_API_KEY)) {
+      this.fulfillRequest(list, res.statusCode, res.body.error);
+      return;
+    }
```

Comment:
Looks like you can merge this part with the above conditional check.

Reply:
resolved